SU2_CFD/src/interfaces/cht/CConjugateHeatInterface.cpp

```diff
@@ -104,95 +93,112 @@ void CConjugateHeatInterface::GetDonor_Variable(CSolver *donor_solution, CGeomet
-    Tnormal = donor_solution->node[PointNormal]->GetPrimitive(0)*Temperature_Ref;
+    Twall   = donor_solution->node[Point_Donor]->GetPrimitive(0);
+    Tnormal = donor_solution->node[PointNormal]->GetPrimitive(0);
 
     dTdn = (Twall - Tnormal)/dist;
   }
   else if (incompressible_flow) {
 
-    Twall   = donor_solution->node[Point_Donor]->GetTemperature()*Temperature_Ref;
-    Tnormal = donor_solution->node[PointNormal]->GetTemperature()*Temperature_Ref;
+    Twall   = donor_solution->node[Point_Donor]->GetTemperature();
+    Tnormal = donor_solution->node[PointNormal]->GetTemperature();
 
     dTdn = (Twall - Tnormal)/dist;
   }
-  else if (flow || heat_equation) {
-    Twall   = donor_solution->node[Point_Donor]->GetSolution(0)*Temperature_Ref;
-    Tnormal = donor_solution->node[PointNormal]->GetSolution(0)*Temperature_Ref;
+  else if (heat_equation) {
+
+    Twall   = donor_solution->node[Point_Donor]->GetSolution(0);
+    Tnormal = donor_solution->node[PointNormal]->GetSolution(0);
 
+    // TODO: Check if these improve accuracy, if needed at all
     //    for (iDim = 0; iDim < nDim; iDim++) {
     //      dTdn += (Twall - Tnormal)/dist * (Edge_Vector[iDim]/dist) * (Normal[iDim]/Area);
     //    }
 
     dTdn = (Twall - Tnormal)/dist;
   }
   else {
-    cout << "WARNING: Transfer of conjugate heat variables is called with non-supported donor solver!" << endl;
+
+    SU2_MPI::Error("Transfer of conjugate heat variables failed (non-supported donor solver).", CURRENT_FUNCTION);
   }
 
-  /*--- Calculate the heat flux density (temperature gradient times thermal conductivity)
-   *--- and set it as second donor variable ---*/
+  /*--- Calculate the heat flux density (temperature gradient times thermal conductivity) ---*/
+
   if (compressible_flow) {
 
-    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
+    su2double Gamma         = donor_config->GetGamma();
+    su2double Gas_Constant  = donor_config->GetGas_ConstantND();
+    su2double Cp            = (Gamma / (Gamma - 1.0)) * Gas_Constant;
+
+    Prandtl_Lam             = donor_config->GetPrandtl_Lam();
+    laminar_viscosity       = donor_solution->node[Point_Donor]->GetLaminarViscosity(); // TDE check for consistency
+    Cp                      = (Gamma / (Gamma - 1.0)) * Gas_Constant;
 
     thermal_conductivityND  = Cp*(laminar_viscosity/Prandtl_Lam);
-    thermal_conductivity    = thermal_conductivityND*donor_config->GetViscosity_Ref();
+    heat_flux_density       = thermal_conductivityND*dTdn;
+
+    if (donor_config->GetCHT_Robin()) {
 
-    heat_flux_density       = thermal_conductivity*dTdn;
-    conductivity_over_dist  = thermal_conductivity/dist;
+      thermal_conductivity    = thermal_conductivityND*donor_config->GetViscosity_Ref();
+      conductivity_over_dist  = thermal_conductivity/dist;
+    }
   }
   else if (incompressible_flow) {
 
-    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
```

Comment:
@oleburghardt just in the process of merging I think this line was removed by mistake? iPoint is used right after

Reply:
Yes, it's a mistake. If you are in the merge window right now, can you revert that change/add that line in your branch? That should be the quickest solution to fix it in develop. Thanks in advance..